music_assistant/providers/pandora/provider.py

```diff
@@ -0,0 +1,432 @@
+                return web.Response(status=404, text="No audio URL available")
+
+            # Update metadata if we have queue context
+            #            if queue_id and queue_item_id:
+            #                queue_item = self.mass.player_queues.get_item(queue_id, queue_item_id)
+            #                if queue_item and queue_item.streamdetails:
+            #                    # Get the best quality album art
+            #                    album_art_url = None
+            #                    if album_art := track.get("albumArt"):
+            #                        # Get the 500px version (good balance of quality/size)
+            #                        album_art_url = next(
+            #                           (art["url"] for art in album_art if art.get("size") == 500),
+            #                           album_art[-1]["url"] if album_art else None,
+            #                        )
+            #                    queue_item.streamdetails.stream_metadata = StreamMetadata(
+            #                        title=track.get("songTitle", "Unknown Song"),
+            #                        artist=track.get("artistName", "Unknown Artist"),
+            #                        album=track.get("albumTitle"),
+            #                        image_url=album_art_url,
+            #                        duration=track.get("trackLength"),
+            #                    )
+            # Redirect to the actual audio URL
+            return web.Response(status=302, headers={"Location": audio_url})
+
+        except Exception as err:
+            self.logger.error(
+                "Error handling stream request for track %d: %s",
+                track_num,
+                err,
+            )
+            return web.Response(status=500, text="A stream error occurred")
+
+    async def search(
+        self,
+        search_query: str,
+        media_types: list[MediaType],
+        limit: int = 25,
+    ) -> SearchResults:
+        """Search library radio stations by name."""
+        # We can't search Pandora's API without the legacy endpoints,
+        # but we can search the user's existing library stations
+        if MediaType.RADIO not in media_types:
+            return SearchResults()
+
+        results: list[Radio] = []
+
+        async for station in self.get_library_radios():
+            if compare_strings(station.name, search_query):
+                results.append(station)
+                if len(results) >= limit:
+                    break
+
+        return SearchResults(radio=results)
+
+    # TODO: Remove this when default implementation is fixed
+    async def browse(self, path: str) -> list[Radio]:
+        """Browse radio stations."""
+        # For now, just return all stations like get_library_radios
+        # Could be enhanced with categories/genres in the future
+        return [station async for station in self.get_library_radios()]
```

Comment:
```suggestion
```

Reply:
I'm not sure yet what needs to change. The only thing I have noticed so far from looking is this line

https://github.com/music-assistant/server/blob/a478337872908123863b6f6b370972451dac0177/music_assistant/models/music_provider.py#L563

should it be `provider=self.instance_id,` like all the others? Not a factor here but I wanted to mention it before I forgot. Beyond that it isn't clear to me why it is failing to render the list?

Reply:
yeah that should be instance_id for sure

Reply:
But what i meant is that we need to fix core - not perse in this PR.
Just remove the workaround so we can merge it and then focus on the issue in core